Gems/ScriptCanvas/Code/Editor/Include/ScriptCanvas/Assets/ScriptCanvasFileHandling.h

```diff
@@ -21,38 +24,22 @@ namespace AZ
 namespace ScriptCanvas
 {
     class ScriptCanvasData;
-}
-
-namespace ScriptCanvasEditor
-{
-    class EditorAssetTree
-    {
-    public:
-        AZ_CLASS_ALLOCATOR(EditorAssetTree, AZ::SystemAllocator, 0);
-
-        EditorAssetTree* m_parent = nullptr;
-        AZStd::vector<EditorAssetTree> m_dependencies;
-        SourceHandle m_asset;
 
-        EditorAssetTree* ModRoot();
+    AZ::Outcome<SourceTree, AZStd::string> LoadEditorAssetTree(SourceHandle handle, SourceTree* parent = nullptr);
 
-        void SetParent(EditorAssetTree& parent);
-
-        AZStd::string ToString(size_t depth = 0) const;
-    };
-
-    // both success and failure return JSON deserialization errors, regardless of parsing / loading success
-    AZ::Outcome<AZStd::string, AZStd::string> LoadDataFromJson
-        ( ScriptCanvas::ScriptCanvasData& dataTarget
-        , AZStd::string_view source
-        , AZ::SerializeContext& serializeContext);
+    struct FileLoadResult
+    {
+        bool m_isSuccess;
+        SourceHandle m_handle;
+        DeserializeResult m_deserializeResult;
+        AZStd::string m_fileReadErrors;
 
-    AZ::Outcome<EditorAssetTree, AZStd::string> LoadEditorAssetTree(SourceHandle handle, EditorAssetTree* parent = nullptr);
+        inline operator bool() const
+        {
+            return m_isSuccess && m_deserializeResult.m_isSuccessful;
+        }
```

Comment:
FYI the `inline` keyword isn't needed here.
It is only needed for an `out(of)line` definition